src/Parallafka/CommitState.cs

```diff
@@ -63,65 +63,22 @@ public CommitState(int maxMessagesQueued, CancellationToken stopToken)
-        public bool TryGetMessageToCommit(IKafkaMessage<TKey, TValue> message, [NotNullWhen(true)] out IKafkaMessage<TKey, TValue> messageToCommit)
-        {
-            messageToCommit = null;
-            Queue<IKafkaMessage<TKey, TValue>> messagesInPartition;
-
-            this._messagesNotYetCommittedByPartitionReaderWriterLock.EnterReadLock();
-            try
-            {
-                if (!this._messagesNotYetCommittedByPartition.TryGetValue(message.Offset.Partition, out messagesInPartition))
-                {
-                    Parallafka<TKey, TValue>.WriteLine($"CS:GetMsgToCommit: {message.Key} {message.Offset} [none]");
-                    return false;
-                }
-            }
-            finally
-            {
-                this._messagesNotYetCommittedByPartitionReaderWriterLock.ExitReadLock();
-            }
-
-            messageToCommit = GetMessageToCommit(message,messagesInPartition);
-            return messageToCommit != null;
-        }
-
-        private IKafkaMessage<TKey, TValue> GetMessageToCommit(IKafkaMessage<TKey, TValue> message, Queue<IKafkaMessage<TKey, TValue>> messagesInPartition)
+        private IKafkaMessage<TKey, TValue> GetMessageToCommit(ConcurrentQueue<IKafkaMessage<TKey, TValue>> messagesInPartition)
         {
             IKafkaMessage<TKey, TValue> messageToCommit = null;
-            lock (messagesInPartition)
+            while (messagesInPartition.TryPeek(out IKafkaMessage<TKey, TValue> msg) && msg.WasHandled)
             {
-                while (messagesInPartition.TryPeek(out IKafkaMessage<TKey, TValue> msg) &&
-                       msg.WasHandled &&
-                       (message == null || msg.Offset.Offset <= message.Offset.Offset))
-                {
-                    messageToCommit = msg;
-                    messagesInPartition.Dequeue();
+                messageToCommit = msg;
+                messagesInPartition.TryDequeue(out _);
 
-                    this._canQueueMessage.Release();
+                this._canQueueMessage.Release();
 
-                    Parallafka<TKey, TValue>.WriteLine($"CS:DequeueMessage: {msg.Key} {msg.Offset}");
-                }
+                Parallafka<TKey, TValue>.WriteLine($"CS:DequeueMessage: {msg.Key} {msg.Offset}");
+            }
 
-                if (message == null)
-                {
-                    Parallafka<TKey, TValue>.WriteLine(
-                        $"CS:GetMsgToCommit: [any] {(messageToCommit == null ? "[notfound]" : messageToCommit.Offset)}");
-                }
-                else
-                {
-                    Parallafka<TKey, TValue>.WriteLine(
-                        $"CS:GetMsgToCommit: {message.Key} {message.Offset} {(messageToCommit == null ? "[notfound]" : messageToCommit.Offset)}");
-                }
+            Parallafka<TKey, TValue>.WriteLine($"CS:GetMsgToCommit: {(messageToCommit == null ? "[notfound]" : messageToCommit.Offset)}");
```

Comment:
I wonder if/when it becomes worth it to have WriteLine take a string-returning function instead

Reply:
WriteLine is wrong long term, just a quick way to get logs showing during unit tests.